UI/NpcList/NpcList.gd

```diff
@@ -1,27 +1,124 @@
-	if(container != null):
-		for child in container.get_children():
-			container.remove_child(child)
-			child.queue_free()
-
 # add pop-up confirmation window
 func forgetNPC(ID):
 	GM.main.removeDynamicCharacter(ID)
 
+
+func _on_Name_pressed():
+	nameButton.pressed = true
+	genderButton.pressed = false
+	personalityButton.pressed = false
+	childrenButton.pressed = false
+	
+	var nodesSortedArr = container.get_children()
+	
+	if(_nameBtnState):
+		nodesSortedArr.sort_custom(self, "sortNameAscending")
+	else:
+		nodesSortedArr.sort_custom(self, "sortNameAscending")
+		nodesSortedArr.invert()
+		
+	for nodeNum in nodesSortedArr.size():
+		container.move_child(nodesSortedArr[nodeNum], nodeNum)
+	
+	_nameBtnState = !_nameBtnState
+
+
+func sortNameAscending(a: Node, b: Node):
+	return a.getNpcName().naturalnocasecmp_to(b.getNpcName()) < 0
+
+
+func _on_Gender_pressed():
+	nameButton.pressed = false
+	genderButton.pressed = true
+	personalityButton.pressed = false
+	childrenButton.pressed = false
+	
+	var nodesSortedArr = container.get_children()
+	
+	if(_genderBtnState):
+		nodesSortedArr.sort_custom(self, "sortGenderAscending")
+	else:
+		nodesSortedArr.sort_custom(self, "sortGenderAscending")
+		nodesSortedArr.invert()
+		
+	for nodeNum in nodesSortedArr.size():
+		container.move_child(nodesSortedArr[nodeNum], nodeNum)
+	
+	_genderBtnState = !_genderBtnState
+
+
+func sortGenderAscending(a: Node, b: Node):
+	return a.getNpcGender().naturalnocasecmp_to(b.getNpcGender()) < 0
+
+
+func _on_Personality_pressed():
+	nameButton.pressed = false
+	genderButton.pressed = false
+	personalityButton.pressed = true
+	childrenButton.pressed = false
+	
+	var nodesSortedArr = container.get_children()
+	
+	if(_personalityBtnState):
+		nodesSortedArr.sort_custom(self, "sortPersonalityAscending")
+	else:
+		nodesSortedArr.sort_custom(self, "sortPersonalityAscending")
+		nodesSortedArr.invert()
+		
+	for nodeNum in nodesSortedArr.size():
+		container.move_child(nodesSortedArr[nodeNum], nodeNum)
+	
+	_personalityBtnState = !_personalityBtnState
+
+
+func sortPersonalityAscending(a: Node, b: Node):
+	return a.getSubbyStatRaw() < b.getSubbyStatRaw()
+
+
+func _on_ChildrenAmount_pressed():
+	nameButton.pressed = false
+	genderButton.pressed = false
+	personalityButton.pressed = false
+	childrenButton.pressed = true
+	
+	var nodesSortedArr = container.get_children()
+	
+	if(_childrenBtnState):
+		nodesSortedArr.sort_custom(self, "sortChildrenAmountAscending")
+	else:
+		nodesSortedArr.sort_custom(self, "sortChildrenAmountAscending")
+		nodesSortedArr.invert()
+		
+	for nodeNum in nodesSortedArr.size():
+		container.move_child(nodesSortedArr[nodeNum], nodeNum)
+	
+	_childrenBtnState = !_childrenBtnState
+
+
+func sortChildrenAmountAscending(a: Node, b: Node):
+	return a.getAmountOfChildren() < b.getAmountOfChildren()
+
+
+func _notification(what):
+	if(what == NOTIFICATION_PREDELETE && GM.main != null):
+		GM.main.playAnimation(StageScene.Solo, "stand", {npc=GM.pc})
```

Comment:
This is way too complicated, just make it so the animation is played when you back out of the menu (make the back button look like `addButton("Back", "Go back a level", "closenpclist")` and then add a new check in the _react function)
```
if(_action == "closenpclist"):
	setState("")
	GM.main.playAnimation(StageScene.Solo, "stand")
	return
```

Reply:
Looks complicated but it is a simple destructor/finalizer, works fine

Reply:
I'd rather you change it still

Reply:
godot doesn't really have destructors, what you're using here is a very low level function. If you don't like my way, I'd like to hear why ^^. Putting the logic inside the scene instead of UI is better, ideally UI shouldn't even know about GM.main but that's hard to achieve in all cases. Still, better to try ^^

Reply:
Honestly I do not like your buttons, every click is a roller-coaster through the half of game code, everything is recalculated like buffs etc I saw in the debugger. It belongs here as an npclist function, works well. GM.main needed to check if the game is still running, I can put anything else here to check, I just don't know what. So that part could be modified.

Reply:
Actually it works even without GM.main check, but throws error in the output window if you just close the game with that menu open and I didn't like that.

Reply:
if you don't like how the buttons work you will have a bad time in the future because the whole game runs on those buttons x3. I tried to only recalculate buffs/status effects only when they need to, trust me. It never worked well, sometimes calculating all that stuff more than once, sometimes not recalculating it when needed. So the best approach that I found is.. doing it once every time the user picks an option, works in all cases even if not the most optimized solution (fine for a text game x3)

But yeah, then I suggest just removing it. It's fine if the animation doesn't resest, it will do that the next time the player moves or something. As you can see you run into problems while using this notification function, who knows how else it's gonna break

Reply:
I didn't run into any actual problems with notification and it is secluded anyway, only possible to be executed while in the characters menu. And it is actually called "destructor" in the official Godot documentation: 
![destructor](https://github.com/Alexofp/BDCC/assets/132950864/a0fc9647-13c6-461f-bd2f-dbdd9117fdd2)

Reply:
You can't imagine how many unique godot quirks I've ran into x3

I trust you, your code works, I'm just trying to make it cleaner ^^

At the end of the day someone gotta have the final say, otherwise we will never stop arguing. Do you wanna have the final say in this project? x3

Reply:
And I thank you for that! 90% of times I fully agree on suggested changes.